kyo-core/shared/src/main/scala/kyo/Adder.scala

```diff
@@ -13,56 +13,75 @@ class LongAdder private[kyo] (private val ref: j.LongAdder) extends AnyVal:
       * @return
       *   Unit
       */
-    inline def add(v: Long)(using Frame): Unit < IO = IO(ref.add(v))
+    inline def add(v: Long)(using inline frame: Frame): Unit < IO = IO(unsafe.add(v))
 
     /** Decrements the sum by one.
       *
       * @return
       *   Unit
       */
-    inline def decrement(using Frame): Unit < IO = IO(ref.decrement())
+    inline def decrement(using inline frame: Frame): Unit < IO = IO(unsafe.decrement())
 
     /** Increments the sum by one.
       *
       * @return
       *   Unit
       */
-    inline def increment(using Frame): Unit < IO = IO(ref.increment())
+    inline def increment(using inline frame: Frame): Unit < IO = IO(unsafe.increment())
 
     /** Returns the current sum.
       *
       * @return
       *   The current sum
       */
-    inline def get(using Frame): Long < IO = IO(ref.sum())
+    inline def get(using inline frame: Frame): Long < IO = IO(unsafe.get())
 
     /** Resets the sum to zero.
       *
       * @return
       *   Unit
       */
-    inline def reset(using Frame): Unit < IO = IO(ref.reset())
+    inline def reset(using inline frame: Frame): Unit < IO = IO(unsafe.reset())
 
     /** Returns the current sum and resets it to zero.
       *
       * @return
       *   The sum before reset,
       */
-    inline def sumThenReset(using Frame): Long < IO = IO(ref.sumThenReset())
+    inline def sumThenReset(using inline frame: Frame): Long < IO = IO(unsafe.sumThenReset())
+
 end LongAdder
 
 object LongAdder:
+
     /** Creates a new LongAdder instance.
       *
       * @return
       *   A new LongAdder
       */
-    def init(using Frame): LongAdder < IO = IO(LongAdder(new j.LongAdder))
+    def init(using frame: Frame): LongAdder < IO = IO(LongAdder(Unsafe.init()))
+
+    opaque type Unsafe = j.LongAdder
+
+    object Unsafe:
+        given Flat[Unsafe]                    = Flat.unsafe.bypass
+        def init()(using AllowUnsafe): Unsafe = new j.LongAdder
```

Comment:
I haven't added scaladocs for unsafe APIs because the source files are getting large but I can add them if we judge it's necessary

Reply:
Unsafe in this context is an operation that can break referential transparency, which is the case for the initialization of an object containing mutable state. I think the mental model should be that these are advanced low-level APIs meant only for scenarios like libraries where performance is a major concern or integrations with other solutions. 

>  Perhaps just a boilerplate on object Unsafe that says something to the effect that every function in there has a side effect so only use if necessary and within IO.apply would be good enough?

Do you mean that `AllowUnsafe` isn't necessary? I think having the extra protection is important so newcomers aren't tempted to use these APIs.

Reply:
I had missed the context from my initial comment 🤦🏽 thanks for clarifying! I've added a top-level warning scaladoc to all unsafe APIs.